packages/react/types/jsx-namespace.d.ts

```diff
@@ -18,7 +18,12 @@ type ReactJSXIntrinsicAttributes = JSX.IntrinsicAttributes
 type ReactJSXIntrinsicClassAttributes<T> = JSX.IntrinsicClassAttributes<T>
 type ReactJSXIntrinsicElements = JSX.IntrinsicElements
 
+// based on the code from @types/react@18.2.8
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/index.d.ts#LL3204C13-L3204C13
+type ReactJSXElementType = string | React.JSXElementConstructor<any>
```

Comment:
Hm, likely we could do some hacks with `@ts-ignore` to "detect" if the type is there in React's `JSX` but perhaps it's not worth it and copy-paste is fine. I'll give this a thought.

Reply:
yeah, I feel like copying it is probably the right approach tbh